Doc/library/tempfile.rst

```diff
@@ -346,6 +381,22 @@ Here are some examples of typical usage of the :mod:`tempfile` module::
     >>>
     # file is now closed and removed
 
+    # create a temporary file using a context manager, note the name,
+    # close the file, use the name to open the file again
+    >>> with tempfile.TemporaryFile(delete_on_close=False) as fp:
+    ...     fp.write(b'Hello world!')
+    ...     fp_name = fp.name
+    ...     fp.close()
+    # the file is closed, but not removed
+    # open the file again by using its name
+    ...     f = open(fp_name)
```

Comment:
`fp.name` is still valid here, right? Why capture it?

Reply:
Also, maybe use this one in its own nested `with` block?

Reply:
I captured the `fp.name` to make more explicit, but I agree, it is redundant, so I removed it now.
Also nested `with` block is added.